pkg/cloud/provider.go

```diff
@@ -517,7 +518,7 @@ func getClusterProperties(node *v1.Node) clusterProperties {
 		accountID:      "",
 		projectID:      "",
 	}
-	if metadata.OnGCE() {
+	if metadata.OnGCE() || strings.HasPrefix(providerID, "gce") {
```

Comment:
This change, in particular, needs review to ensure it's appropriate. `metadata.OnGCE()` internally uses `sync.Once`, which means, in the context of `cloud_test.go`, either you're on GCP for all the tests or you're not on GCP for any of the tests.

I don't know the history of why this condition uses `metadata.OnGCE()`, instead of looking at the provider ID prefix, as the other conditions do. I know next to nothing about GCP or what its provider IDs would look like in KubeCost. Based on some other things--in particular, `providerGCERegex` in `provider.go`--I think the `gce` prefix is appropriate. If this is OK, is `metadata.OnGCE()` needed at all?

Reply:
From my understanding, this merely an "official" GCP client library way to determine this, and is therefore more advisable to use than the prefix check.

Is it possible to work around the testing limitation? Perhaps change `MetadataHostEnv` in the tests to reflect the provider that should be used (see [here](https://github.com/googleapis/google-cloud-go/blob/601c77a69a27b5f13ebb4508f8222a98c8a904bc/compute/metadata/metadata.go#L114-L189))?